packages/core/test/acceptance/pending_tasks_spec.ts

```diff
@@ -113,6 +113,21 @@ describe('public PendingTasks', () => {
     await expectAsync(appRef.whenStable()).toBeResolved();
     expect(spy).toHaveBeenCalled();
   });
+
+  it('should allow calling cleanup function multiple times', async () => {
+    const appRef = TestBed.inject(ApplicationRef);
+    const pendingTasks = TestBed.inject(PendingTasks);
+
+    const removeTask = pendingTasks.add();
+    await expectAsync(applicationRefIsStable(appRef)).toBeResolvedTo(false);
+
+    removeTask();
+    removeTask();
+    removeTask();
+
+    TestBed.inject(ApplicationRef).tick();
+    await expectAsync(applicationRefIsStable(appRef)).toBeResolvedTo(true);
+  });
```

Comment:
Currently this test does not do anything related to the existence/use of `Symbol.dispose`.  You can remove the new code and the test passes without issue.

For a test you would need to confirm that once the block that the the `cleanup` symbol is defined within is exited, then the dispose method should automatically be called.  I think what you are looking for is a test like this:


```suggestion
  it('should automatically cleanup the pending task once its declaration block is exited', async () => {
    const appRef = TestBed.inject(ApplicationRef);
    const pendingTasks = TestBed.inject(PendingTasks);

    {
      using removeTask = pendingTasks.add();
      await expectAsync(applicationRefIsStable(appRef)).toBeResolvedTo(false);
    }

    TestBed.inject(ApplicationRef).tick();
    await expectAsync(applicationRefIsStable(appRef)).toBeResolvedTo(true);
  });
```